### PR TITLE
Minor: Refactor: Re-use existing IsRootless() function

### PR DESCRIPTION
**Pull request by isimluk**
Signed-off-by: Šimon Lukašík <isimluk@fedoraproject.org>

### CODE REVIEW

File: pkg/rootless/rootless_linux.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func tryMappingTool(tool string, pid int, hostID int, mappings []idtools.IDMap)
 // JoinNS re-exec podman in a new userNS and join the user namespace of the specified
 // PID.
 func JoinNS(pid uint) (bool, int, error) {
-	if os.Geteuid() == 0 || os.Getenv("_LIBPOD_USERNS_CONFIGURED") != "" {
```
Comment (giuseppe) [inline pkg/rootless/rootless_linux.go]: here we are checking if `os.Geteuid() == 0` in the `IsRootless()` function we check for `os.Geteuid() != 0`, so it is not equivalent.

File: pkg/rootless/rootless_linux.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ func JoinNS(pid uint) (bool, int, error) {
 // JoinNSPath re-exec podman in a new userNS and join the owner user namespace of the
 // specified path.
 func JoinNSPath(path string) (bool, int, error) {
-	if os.Geteuid() == 0 || os.Getenv("_LIBPOD_USERNS_CONFIGURED") != "" {
```
Comment (giuseppe) [inline pkg/rootless/rootless_linux.go]: same here

File: pkg/rootless/rootless_linux.go [hunk 3]
```diff
@@ -159,7 +159,7 @@ func getMinimumIDs(p string) int {
 // If podman was re-executed the caller needs to propagate the error code returned by the child
 // process.
 func BecomeRootInUserNS() (bool, int, error) {
-	if os.Geteuid() == 0 || os.Getenv("_LIBPOD_USERNS_CONFIGURED") != "" {
```
Comment (giuseppe) [inline pkg/rootless/rootless_linux.go]: and here

File: pkg/rootless/rootless_linux.go (hunk 1 above)
Comment (isimluk) [reply pkg/rootless/rootless_linux.go]: Oh man 🙈 am I blind or something? ...sorry... and nice catch!

Comment (giuseppe) [reply pkg/rootless/rootless_linux.go]: no problem!  I also needed to double check it "why wasn't this refactored?"

**Pull request closed without merging**